### ISSUE TITLE
List of folders not populated when using External Storage

### ISSUE THREAD

**Issue by fedev** [owncloud/client#3740]
Hi,

So I've been using ownCloud for a couple of years, everything was going great until the last server/client upgrade. Now using:

Server: 8.1.1-12.1 (OpenSuSe)
Client: 2.0.0-1 (Arch Linux)

When External Storage is enabled, synchronization fails. If I try to setup the application again, the list of folders to be synced does not show up. It will stay on "Loading..." and then error out.

The logs during this time show:
```
09-03 08:16:12:129 10116 OCC::OwncloudWizard::slotCurrentPageChanged: Current Wizard page changed to  3
09-03 08:16:14:636 10116 OCC::AbstractNetworkJob::start: !!! OCC::LsColJob created for "https://hostname" + ""
09-03 08:16:56:941 10116 OCC::SocketApi::slotNewConnection: SocketApi:  New connection QLocalSocket(0x177ce90)
09-03 08:17:13:376 10116 OCC::SocketApi::onLostConnection: SocketApi:  Lost connection  QLocalSocket(0x177ce90)
09-03 08:21:18:495 10116 OCC::AbstractNetworkJob::slotTimeout: virtual void OCC::AbstractNetworkJob::slotTimeout() OCC::LsColJob(0x198f8a0) Timeout QUrl("https://hostname:54443/remote.php/webdav/")
09-03 08:21:18:495 10116 OCC::AbstractNetworkJob::slotFinished: void OCC::AbstractNetworkJob::slotFinished() QNetworkReply::NetworkError(OperationCanceledError) "Operation canceled" QVariant(Invalid)
```

Once External Storage is disabled, the process continues successfully as you can see in the logs:

```
09-03 08:24:35:373 10116 OCC::OwncloudWizard::slotCurrentPageChanged: Current Wizard page changed to  3
09-03 08:24:48:713 10116 OCC::OwncloudSetupWizard::slotCreateLocalAndRemoteFolders: Setup local sync folder for new oC connection  "/home/user/ownCloud"
09-03 08:24:48:713 10116 OCC::OwncloudWizard::appendToConfigurationLog: Setup-Log:  "Local sync folder /home/user/ownCloud already exists, setting it up for sync.<br/><br/>"
09-03 08:24:48:714 10116 OCC::AbstractNetworkJob::start: !!! OCC::EntityExistsJob created for "https://hostname" + "remote.php/webdav//"
09-03 08:24:51:378 10116 OCC::OwncloudSetupWizard::slotRemoteFolderExists: ******** Remote folder found, all cool!
```

I'm not sure why it is failing to access the External Storage. I must say this issue affects only the Desktop Client. The web client and the mobile client work as expected.

Thanks!

Regards,

Federico

**Comment by phil-davis** [comment]
I changed the title of https://github.com/owncloud/client/issues/3524 to better reflect what turned out to be the root cause. I think this and that are the same issue.

**Comment by ogoffart** [comment]
It looks like a timeout.
Could it be that the server is too slow to anser the requests?
Is there any log on the server that show that the request was successfull?

**Comment by guruz** [comment]
...and which type of external storage is it?
what does ```owncloud.log``` on the server say?
Did the client or the server upgrade cause it?

**Comment by fedev** [comment]
The OwnCloud server is running in a headless VM (qemu). The external drive is a samba share hosted on the server running the VM. Owncloud has a filter set not to synchronize automatically anything from that share.

This was working before on the desktop client. On the mobile client this still works. The only way to get the desktop client to work is to disable the external storage.

The SMB logs for the OwnCloud host show:
```
[2015/09/03 08:14:56.858796,  2] ../source3/smbd/service.c:862(make_connection_snum)
  owncloud (ipv4:10.xxx.xxx.xxx:56586) connect to service mapdrives initially as user nobody (uid=65534, gid=65533) (pid 23198)
[2015/09/03 08:18:42.192949,  2] ../source3/smbd/service.c:862(make_connection_snum)
  owncloud (ipv4:10.xxx.xxx.xxx:56601) connect to service mapdrives initially as user nobody (uid=65534, gid=65533) (pid 23329)
[2015/09/03 08:29:56.788802,  2] ../source3/smbd/process.c:2779(deadtime_fn)
  Closing idle connection
[2015/09/03 08:29:56.789034,  2] ../source3/smbd/service.c:1138(close_cnum)
  owncloud (ipv4:10.xxx.xxx.xxx:56586) closed connection to service mapdrives
```

I couldn't find anything unusual in owncloud.log.

**Comment by cleocir** [comment]
Same problem here.
When i enable external storage (other owncloud server) i get the same errors.

**Comment by guruz** [comment]
Please check
https://github.com/owncloud/documentation/blob/master/admin_manual/enterprise_external_storage/windows-network-drive_configuration.rst
https://owncloud.org/support

**Comment by fedev** [comment]
Just to note that the above was not really relevant for me. The issue was fixed once upgraded to 8.2.2.2.
